db/migrations/20201220213114_create_circle_basic_schema.js

```diff
@@ -0,0 +1,217 @@
+
+    // foreign keys
+    t.foreign('circle_id')
+      .references('id')
+      .inTable('circle')
+      .onUpdate('CASCADE')
+      .onDelete('CASCADE')
+  })
+
+  // table for article - circle relationship
+  table = 'article_circle'
+
+  await knex('entity_type').insert({ table })
+  await knex.schema.createTable(table, (t) => {
+    t.bigIncrements('id')
+    t.bigInteger('article_id').unsigned().notNullable()
+    t.bigInteger('circle_id').unsigned().notNullable()
+    t.bigInteger('creator').unsigned().notNullable()
+    t.timestamp('created_at').defaultTo(knex.fn.now())
+    t.timestamp('updated_at').defaultTo(knex.fn.now())
+
+    // foreign keys
+    t.foreign('article_id')
+      .references('id')
+      .inTable('article')
+      .onUpdate('CASCADE')
+      .onDelete('CASCADE')
+    t.foreign('circle_id')
+      .references('id')
+      .inTable('circle')
+      .onUpdate('CASCADE')
+      .onDelete('CASCADE')
+    t.foreign('creator')
+      .references('id')
+      .inTable('user')
+      .onUpdate('CASCADE')
+      .onDelete('SET NULL')
+  })
+
+  // table for user - subscription relationship
+  table = 'circle_subscription'
+
+  await knex('entity_type').insert({ table })
+  await knex.schema.createTable(table, (t) => {
+    t.bigIncrements('id')
+    t.enu('state', ['active', 'archived'])
+    t.bigInteger('user_id').unsigned().notNullable()
+    t.timestamp('created_at').defaultTo(knex.fn.now())
+    t.timestamp('updated_at').defaultTo(knex.fn.now())
+    t.timestamp('canceled_at')
+    t.enu('provider', ['stripe']).notNullable().defaultTo('stripe')
+    t.string('provider_subscription_id').notNullable().unique()
+
+    // foreign keys
+    t.foreign('user_id')
+      .references('id')
+      .inTable('user')
+      .onUpdate('CASCADE')
+      .onDelete('CASCADE')
+  })
+
+  // table for subscription - price relationship
+  table = 'subscription_price'
+
+  await knex('entity_type').insert({ table })
+  await knex.schema.createTable(table, (t) => {
+    t.bigIncrements('id')
+    t.enu('state', ['active', 'archived'])
+    t.bigInteger('creator').unsigned().notNullable()
+    t.bigInteger('subscription_id').unsigned().notNullable()
+    t.bigInteger('price_id').unsigned().notNullable()
+    t.timestamp('created_at').defaultTo(knex.fn.now())
+    t.timestamp('updated_at').defaultTo(knex.fn.now())
+
+    // foreign keys
+    t.foreign('creator')
+      .references('id')
+      .inTable('user')
+      .onUpdate('CASCADE')
+      .onDelete('SET NULL')
+    t.foreign('subscription_id')
+      .references('id')
+      .inTable('circle_subscription')
+      .onUpdate('CASCADE')
+      .onDelete('CASCADE')
+    t.foreign('price_id')
+      .references('id')
+      .inTable('circle_price')
+      .onUpdate('CASCADE')
+      .onDelete('CASCADE')
+  })
+
+  // update comment table to target circle or article
+  // add columns
+  table = 'comment'
+  await knex.schema.table(table, (t) => {
+    t.bigInteger('target_id').unsigned()
```

Comment:
`target_id ` and `target_type_id` should be `.notNullable()`?

Reply:
Constraints are added [later](https://github.com/thematters/matters-server/pull/1664/files#diff-86288cb77645b5bc4f723d086767d49b6c583ae98433a8fa83a58c2e3ac7b59bR200) since we need to do migration to fill in data first.

Reply:
And how do we know comments belong to broadcast or discussion?

Reply:
Good point, I think the simplest way is adding a `type` field 🤔